### PR TITLE
Add explicit x-internal tags to all schemas

### PR DESCRIPTION
**Pull request by PragalvaXFREZ**
**Notes for Reviewers**
Previously, omitting `x-internal tags` implied that the schema applied to all downstream consumers, making the behavior implicit instead of explicit.

This PR updates all such shared schemas to define x-internal tags explicitly.

Shared schemas are now annotated as:
`x-internal: ["cloud", "meshery"]`

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: validation/rules_codegen.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func checkRule11(filePath string, doc *openapi3.T, _ AuditOptions) []Violation {
 	return out
 }
 
-// --- Rule 14: x-internal must be omitted or ["cloud"]/["meshery"] ---
+// --- Rule 14: x-internal is required on every operation; values must be "cloud" or "meshery" ---
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply validation/rules_codegen.go]: updated in latest commit

File: validation/rules_codegen.go [hunk 2]
```diff
@@ -59,11 +59,14 @@ func checkRule14(filePath string, doc *openapi3.T, _ AuditOptions) []Violation {
...
 			raw, ok := op.Extensions["x-internal"]
 			if !ok {
+				out = append(out, Violation{File: filePath,
+					Message: fmt.Sprintf(`%s — x-internal is required on every operation. Use ["cloud"], ["meshery"], or ["cloud", "meshery"].`, label),
+					Severity: SeverityBlocking, RuleNumber: 14})
 				continue
 			}
-			label := fmt.Sprintf("%s %s", strings.ToUpper(method), path)
 			arr, ok := raw.([]any)
 			if !ok {
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply validation/rules_codegen.go:71]: checkRule14 only sees operations loaded through kin-openapi from api.yml, and in that path x-internal is decoded as []any. The []string case is already handled in parseXInternalTargets for the separate programmatic-input path, so adding it here would be redundant.

**Comment by Junnygram** [comment]
This prevents accidental leakage of internal APIs to public consumers LGTM ..

**Review by Oluwatunmise-olat — APPROVED** [review]
LGTM overall. Since this tightens Rule 14 and bundler behavior, is it possible to add targeted tests for checkRule14 (missing x-internal, empty array, invalid value, valid [`cloud`,`meshery`]) and for filterOpenapiByTag.js fail-fast when x-internal is missing?
